Makefile

```diff
@@ -161,6 +161,24 @@ cover:
 .PHONY: cover
 
 
+tag-and-build:
+	@git diff --quiet || (echo "working directory must be clean"; exit 2)
+	echo ${VERSION} > version.txt
```

Comment:
isn't `VERSION` being read from the `version.txt` file why write it back?

Reply:
See what I wrote about usage. The idea is to overwrite this when calling to change the version number. This is tested and works.

If you have a nicer suggestion, I’m open to it...

Reply:
so what you pass in the command overrides the first line which reads from file ?
I currently don't have a better suggestion :)

Reply:
yes, based on [this answer in SO](https://stackoverflow.com/a/2826178/1225467):

>**From command line** - `make` can take variable assignments as part of his command line, mingled with targets:
>
>```
>make target FOO=bar
>```
>But then all assignments to `FOO` variable within the makefile will be ignored unless you use the override directive in assignment.

(tested and it works)